src/safeds/data/tabular/containers/_table.py

```diff
@@ -355,7 +356,7 @@ def __eq__(self, other: Any) -> bool:
             return NotImplemented
         if self is other:
             return True
-        if self.number_of_rows == 0 and other.number_of_rows == 0:
+        if self.number_of_columns == 0 and other.number_of_columns == 0:
             return self.column_names == other.column_names
```

Comment:
```suggestion
            return true
```
If neither Table has any columns, they are both empty tables and therefore always equals